util/microservice-sdk/js/README.md

```diff
@@ -8,6 +8,132 @@ library scoped to attend the dojot microservices necessities.
 
 ## Usage
 
+### Logging
+
+To log messages, you use the Logger class, which is a wrapper over the winston logging
+library. It supports logging messages to both 'console' and 'file' which is rotated.
+
+The messages are logged in the 'console' using the following format:
+
+```bash
+ts -- sid -- LEVEL: [rid] message (file:line) {extra}
+```
+
+Where:
+
+* ts: timestamp in ISO format
+* sid: service identification
+* rid: request identification (optional metadata)
+* level: error, warn, info or debug
+* [rid]: request identifier (optional metadata)
+* message: message in string format
+* (file:line): file and line of the log message (optional metadata)
+* extra: any additional data - user specific (optional metadada)
+
+The messages are logged in the 'file' using the following format:
+
+```js
+{
+  level: string,
+  timestamp: <number>,
+  sid: <string>,
+  rid: <string>,
+  message: <string>,
+  file: <string>,
+  line: <number>,
+  extra: <object>,
+ }
+```
+
+Where:
+
+* ts: timestamp in unix format
+* sid: service identification
+* rid: request identification (optional metadata)
+* level: error, warn, info or debug
```

Comment:
What happens if these parameters are in upper case or with 2, as inFO?

Reply:
An error will be thrown if you don't use the right case (all lower case).

Reply:
What if we always change the values of the environment variables to lowercase?

Reply:
I didn't get your point. The user will pass the data throw the APIs. S/He will not pass the log level as parameter, but will call the appropriate method.

Reply:
util/microservice-sdk/js/lib/logging/Logger.js at line 59

change to something like that:

```js
  static setLevel(transport, level) {
  
    const levelLower = level.toLowerCase();
    // validate parameters
    if (!isTransportValid(transport)) {
      throw new Error('The transport value is not valid.');
    }
    if (!isLevelValid(levelLower)) {
      throw new Error('The level value is not valid.');
    }

    // set level
    if (wlogger.transports[transport]) {
      wlogger.transports[transport].level = levelLower;
    } else {
      throw new Error('Transport has\'nt been set.');
    }
  }
```
if the user passes INFO, there will be no problems.
I had problems like this in the past, some services only accept upper.

Reply:
Ok.